### ISSUE TITLE
Backspacing the last character in a paragraph hangs on iOS Safari  

### ISSUE THREAD

**Issue by barbatus** [ProseMirror/prosemirror#1271]
### Issue details
Thank you for the prosemirror, it's been great using it so far.

There is a message composer in our app that uses TipTap@2.0.0-beta.174. Some of our users reported an issue some time ago that deleting the last character in a paragraph makes the whole app hangs on iOS specifically.

### Steps to reproduce
There is no clear way how to reproduce it. Say, this [basic code works](https://codesandbox.io/s/kxtm4z) fine on Safari on my iOS simulator but when I try the same code in the app, it starts to hang right away. I assume it depends on the DOM complexity.

I was able to track it down to the place where `<br class="ProseMirror-trailingBreak">` is inserted [here](https://github.com/ProseMirror/prosemirror-view/blob/master/src/viewdesc.js#L744). Say, If I comment `updater.addTextblockHacks()` I am able to delete the last character, another workaround found if I monkey patch `domObserver.flush` to make it run in setTimeout then it works fine again. Please, check out the screenshot video below on how it hangs in the DOM.

I can see that `EditorView` stops and starts observing DOM mutations when it makes its own changes to the DOM, thus I assume it expects that Safari commits DOM changes synchronously but it seems to be not true, otherwise, I can't explain why mutations observer's callback is called with changes made in between `domObserver.stop` and then  `domObserver.start` (if I am not wrong).
### ProseMirror version
prosemirror-view@1.23.6
### Affected platforms
iOS Safari
### Screenshots / Screencast (Optional)
https://user-images.githubusercontent.com/2824096/168818809-6af829c8-29f1-4981-8165-c75240158766.mov

**Comment by marijnh** [comment]
Without a way to reproduce this in plain ProseMirror, it is unlikely that I can figure out what is going wrong (and quite likely that it is something your app is doing). So if you want progress on this, isolating the code needed to reproduce it would be your best bet.

**Comment by natanael** [comment]
An easy way to replicate the issue is to open [the basic example](https://prosemirror.net/examples/basic/) in Safari (MacOS will do, no need to try iOS), run the following in the dev console:
```
window.addEventListener('DOMNodeInserted', () => { for(let i = 0; i < 1e5; i++) { new Date() } })
```
And try to erase a paragraph.

**Comment by marijnh** [comment]
Thanks for the reproduction instructions. Attached patch should help. Apparently Safari starts delivering records for mutations that happened while the observer wasn't observing when a DOMNodeInserted handler is present, which caused the editor to get into a loop where it kept reacting to DOM changes that it itself made.

**Comment by aminland** [comment]
Any word on when there'll be a release that includes this fix?

**Comment by marijnh** [comment]
I've published 1.26.3
